File: 프로그래머스/Level1/신고 결과 받기.py
# https://programmers.co.kr/learn/courses/30/lessons/92334
def solution(id_list, report, k):
    answer = []
    reported_time={}
    report_people={}
    report = list(set(report))
    for j in id_list:
        reported_time[j]=0
        report_people[j]=[]
    for i in report:
        person1, person2 = i.split(' ')
        reported_time[person2]+=1
        report_people[person1].append(person2)
        
    for m in report_people:
        count =0
        for l in report_people[m]:
            if reported_time[l]>=k:
                count+=1
        answer.append(count)
                
    return answer

File: 프로그래머스/Level1/신고 결과 받기.js
// https://programmers.co.kr/learn/courses/30/lessons/92334?language=javascript
function solution(id_list, report, k) {
  let answer = [];
  report = [...new Set(report)];
  // const reported = report.map(el => el.split(' ')[1]);
  let reports = [ ...new Set(report) ]
  let reported_people = {};
  let reporting_count = {};
  id_list.forEach((id) =>{
      reported_people[id] = [];
      reporting_count[id] = 0;
  })
  reports.forEach((repo) =>{
      let [one,two] = repo.split(' ');
      reported_people[two].push(one);
  })
  id_list.forEach((person) => {
      if (reported_people[person].length >=k){
          reported_people[person].forEach((reporting_person) => {
              reporting_count[reporting_person]+=1;
          })
      }
  })
  for(let person in reporting_count){
      answer.push(reporting_count[person]);
  }
  return answer;
}